packages/cmd/src/response/CommandResponse.ts

```diff
@@ -625,9 +626,44 @@ export class CommandResponse implements ICommandResponseApi {
                         return CliUtils.promptWithTimeout(questionText, opts?.hideText, opts?.secToWait);
                     }
                 }
+
+                /**
+                 * Handles interactive menu selection for command input
+                 * @param {string[]} menu
+                 * @param {IInteractiveOptions} [opts]
+                 */
+                public interactiveSelection(menu: string[], opts?: IInteractiveOptions): Promise<number> {
+                    if (outer.mStream) {
+                        return new Promise<number>((resolve) => {
+                            // build interactive header
+                            const daemonHeaders = DaemonUtils.buildHeader({interactive: true});
+
+                            const obj: any = {}
+                            if (opts?.header) obj.header = opts.header;
+                            obj.menu = menu;
+                            outer.writeStream(JSON.stringify(obj));
```

Comment:
Should we create `x-zowe-daemon-data` (and maybe `x-zowe-daemon-data-length`) instead of pushing strings before the headers?

Reply:
this seems like a cleaner way to handle between client and daemon - perhaps this can be done before `next` becomes `latest` ?

Reply:
This should be resolved by https://github.com/zowe/imperative/pull/681